### ISSUE TITLE
Wrong MERGE statement when On always TRUE

### ISSUE THREAD

**Issue by Usaga** [linq2db/linq2db#3589]
Hi!

There is wrong MERGE statement generated when `On` method has no actual predicate.

```c#
        class Settings
        {
            public int Id { get; set; }
        }

        using var ctx = new ArmsContext();

        var mat = new Settings();

        ctx.GetTable<Settings>()
            .Merge()
            .Using(new[] { mat })
            .On((a, b) => true) // Trying to insert or update single (or all, which is not expected) record
            .InsertWhenNotMatched()
            .UpdateWhenMatched()
            .Merge();
```

```sql
MERGE INTO [Settings] [Target]
USING (VALUES
        (0)
) [Source]
(
        [Id]
)
ON (()) -- Expected 1=1, or similar

WHEN NOT MATCHED THEN
INSERT
(
        [Id]
)
VALUES
(
        [Source].[Id]
)

WHEN MATCHED THEN
UPDATE
SET
        [Target].[Id] = [Source].[Id]
;
```

Current workaround I used is:

```C#
.On((a, b) => a.Id != 0)
```

### Environment details
linq2db version: 4.0
Database Server: SQL Server 2012
Database Provider: System.Data.Sql
Operating system: Windows 10
.NET Framework: 4.8

**Comment by viceroypenguin** [comment]
Would `ctx.InsertOrReplace()` work instead?

**Comment by MaceWindu** [comment]
Fixed in `6.0.0-preview.1`
